pkg/cli/httpClient.go

```diff
@@ -0,0 +1,178 @@
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	zotErrors "github.com/anuvu/zot/errors"
+)
+
+var httpClient *http.Client = createHTTPClient()
+
+func createHTTPClient() *http.Client {
+	return &http.Client{
+		Timeout: 5 * time.Second,
+	}
+}
+func makeGETRequest(url, username, password string, resultStructPointer interface{}) (http.Header, error) {
+	attemptRLimitSet()
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.SetBasicAuth(username, password)
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		if resp.StatusCode == http.StatusUnauthorized {
+			return nil, zotErrors.ErrUnauthorizedAccess
+		}
+		bodyBytes, _ := ioutil.ReadAll(resp.Body)
+		return nil, errors.New(string(bodyBytes))
+	}
+
+	if err := json.NewDecoder(resp.Body).Decode(resultStructPointer); err != nil {
+		return nil, err
+	}
+
+	return resp.Header, nil
+}
+
+func attemptRLimitSet() {
```

Comment:
now that we are ratelimiting at client side and being "nice" to the server, we may yet still need this!

Reply:
we can either remove it or set it to some reasonable value instead of MAX